homeassistant/components/workday/binary_sensor.py

```diff
@@ -165,6 +174,23 @@ async def async_setup_entry(
         _holiday_string = holiday_date.strftime("%Y-%m-%d")
         LOGGER.debug("%s %s", _holiday_string, name)
 
+    async def check_date(service_call: ServiceCall) -> ServiceResponse:
+        """Check if date is workday or not."""
+        date_to_test: date = service_call.data[CHECK_DATE]
+        holiday_date = date_to_test in obj_holidays
+
+        date_string = date_to_test.strftime("%Y-%m-%d")
+
+        return {"date": date_string, "workday": not holiday_date}
+
+    hass.services.async_register(
+        DOMAIN,
+        SERVICE_CHECK_DATE,
+        check_date,
+        cv.make_entity_service_schema({vol.Required(CHECK_DATE): cv.date}),
+        SupportsResponse.ONLY,
+    )
```

Comment:
I would expect that domain services are registered in the `__init__.py` file or a dedicated file but not inside a platform file.
Therefore I would suggest moving it into `__init__.py`. What do you think about it?